### ISSUE TITLE
Bluescreens on Azure windows VMs - request debug help

### ISSUE THREAD

**Issue by mweibel** [microsoft/Windows-Containers#342]
**What is your question?**
initially discussed this in [#sig-windows on Kubernetes slack](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1679476637898599) and got asked to post an issue here. 

We're investigating bluescreens which we encounter on azure windows VMs (cluster is CAPZ based).
We run a heavy container (~40GB if I'm not mistaken) containing 3dsmax and plugins inside.

**System environment**
- CAPZ Kubernetes cluster (self managed, not AKS)
- Windows Server 2019 VMs in VMSS (Azure `Standard_D96as_v4` / `Standard_D96as_v5`)
- All VMs are "single" use (fresh VM is boot up upon receiving a `Pending` pod and shut down (if no other Pod is pending) after the job run through - a couple of hours of run time per VM)
- Kubernetes 1.24.11
- containerd 1.6.19 (tried also quickly on-the-fly with 1.7 but didn't change behaviour)
- CNI calico running as service on the hosts (not HostProcess since we had troubles making it stable + quick enough)
- CSI SMB with 3 shares to pods running in the same k8s cluster for data
- VM image is built using packer based on image 2019-Datacenter-Core-smalldisk (we use [image-builder](https://github.com/kubernetes-sigs/image-builder)) just like CAPZ.
- [Argo workflows](https://github.com/argoproj/argo-workflows) with emissary executor for running the jobs

The BSOD seems to most often happen during stopping of the container.
We have minidumps and some memory dump files and tried to analyze them (we're unfortunately not windows experts).

Bugcheck code is 0xCB.
[bluescreen view](http://www.nirsoft.net/utils/blue_screen_view.html) tells it's ntoskrnl.exe crashing.

Trying to investigate more (e.g. using the [howto here](https://github.com/MicrosoftDocs/windows-driver-docs/blob/staging/windows-driver-docs-pr/debugger/interpreting-bug-check-0xcb.md) to figure out what's happening but didn't have much success at this point.

FYI We contacted Azure support about this too - I assume somebody is having a look at this case. 

I'm going to send more details with the minidumps/memory dumps to your email (`@service microsoft.com`). 

Do you have anything which could help us narrow down the issue? Any pointers what we should check?

Thanks!

**Comment by fady-azmy-msft** [comment]
Hey @mweibel , we can't look at issues until we have repro steps ([see our Issues policy](https://github.com/microsoft/Windows-Containers#issues)). Can you share how we can reproduce this issue?

**Comment by mweibel** [comment]
hi @fady-azmy-msft 

thanks for the reply. I understand the policy but it's a bit hard to reproduce as it requires:
- a custom VM image
- a custom container
- a specific invocation and input for the container

Not sure if it makes sense. However we got a bit further meanwhile: we noticed a reboot after init of the node (join k8s cluster, CNi instal/init, and a few other things) helps and seems to avoid the blue screen. We're not quite sure what exactly is requiring a reboot in the init process to we don't know yet how to avoid it, unfortunately. 

Any pointers in debugging would be greatly appreciated!
